### PR TITLE
/last estimated next cat time

### PR DESCRIPTION
**Pull request by MonkeysWithPie**
i had an epiphany that /last could show when the next cat might spawn so that would be pretty neat i think

### CODE REVIEW

**Comment by blaumeise20** [comment]
You should store the next spawn time and display the exact moment it spawns :trell:

**Comment by milenakos** [comment]
> You should store the next spawn time and display the exact moment it spawns :trell:

its stored already in .yet_to_spawn

**Comment by blaumeise20** [comment]
Oh then why not use that? :staring_cat:

**Comment by milenakos** [comment]
bc blaumiese is stupid

**Comment by milenakos** [comment]
truth

**Comment by milenakos** [comment]
dumbomeise20

**Comment by MonkeysWithPie** [comment]
> Oh then why not use that?

it's not public information :SILENCE:

File: main.py [hunk 1]
```diff
@@ -1477,14 +1477,21 @@ async def daily(message: discord.Interaction):
...
-        lasttime = Channel.get_or_none(channel_id=message.channel.id).lastcatches
+        lasttime = channel.lastcatches
         displayedtime = f"<t:{int(lasttime)}:R>"
     except Exception:
-        displayedtime = "forever ago"
-    await message.response.send_message(f"the last cat in this channel was caught {displayedtime}.")
+        displayedtime = "never"
+    
+    if channel and not channel.cat:
+        nextpossible = "\nthe next cat will spawn <t:" + {int(lasttime)+channel.spawn_times_min} + "R:> to <t:" + {int(lasttime)+channel.spawn_times_max} + ":R>"
```
Comment (milenakos) [inline main.py:1492]: buddyy cannot decide between normal strings and f-strings
```suggestion
        nextpossible = f"\nthe next cat will spawn <t:{int(lasttime) + channel.spawn_times_min}R:> to <t:{int(lasttime) + channel.spawn_times_max}:R>"
```

File: main.py [hunk 2]
```diff
@@ -1477,14 +1477,21 @@ async def daily(message: discord.Interaction):
...
+    nextpossible = ""
+
     try:
-        lasttime = Channel.get_or_none(channel_id=message.channel.id).lastcatches
+        lasttime = channel.lastcatches
         displayedtime = f"<t:{int(lasttime)}:R>"
     except Exception:
-        displayedtime = "forever ago"
-    await message.response.send_message(f"the last cat in this channel was caught {displayedtime}.")
+        displayedtime = "never"
```
Comment (milenakos) [inline main.py:1489]: i kinda liked "forever ago" better

**Review by milenakos — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (MonkeysWithPie) [reply main.py:1492]: sorry i'm not very good with snakes

**Comment by MonkeysWithPie** [comment]
ok how do i say that i made the changes correctly it's trying to tell me that i didn't

**Review by milenakos — APPROVED** [review]
lgtm

**Pull request merged**